python/whylogs/experimental/core/metrics/embedding_metric.py

```diff
@@ -166,10 +167,26 @@ def _update_submetrics(self, submetric: str, data: PreprocessedColumn) -> None:
             self.submetrics[submetric][key].columnar_update(data)
 
     def columnar_update(self, data: PreprocessedColumn) -> OperationResult:
-        if data.numpy.ints is None and data.numpy.floats is None:
+        if data.numpy.ints is None and data.numpy.floats is None and data.pandas.vectors is None:
             return OperationResult.ok(0)
 
-        matrices = [X for X in [data.numpy.ints, data.numpy.floats] if X is not None]
+        converted_numpy_ints = None
+        if data.numpy.ints is not None and len(data.numpy.ints) > 0:
+            converted_numpy_ints = data.numpy.ints
+            if is_not_stub(pd.Series) and isinstance(data.numpy.ints, pd.Series) and not data.numpy.ints.empty:
+                converted_numpy_ints = data.numpy.ints.to_numpy()
+
+        converted_numpy_floats = None
+        if data.numpy.floats is not None and len(data.numpy.floats) > 0:
+            converted_numpy_floats = data.numpy.floats
+            if is_not_stub(pd.Series) and isinstance(data.numpy.floats, pd.Series) and not data.numpy.floats.empty:
+                converted_numpy_floats = data.numpy.floats.to_numpy()
+
+        converted_vectors = None
+        if data.pandas.vectors is not None:
+            converted_vectors = np.stack(data.pandas.vectors.to_numpy(), axis=1)
+
+        matrices = [X for X in [converted_numpy_ints, converted_numpy_floats, converted_vectors] if X is not None]
```

Comment:
I think we just want to grab the tensors out of the `ListView` and `PandasView`.

Reply:
This is done.